interp/cfg.go

```diff
@@ -2135,7 +2135,7 @@ func compDefineX(sc *scope, n *node) error {
 	for i, t := range types {
 		var index int
 		id := n.child[i].ident
-		if sym, level, ok := sc.lookup(id); ok && level == n.child[i].level && sym.kind == varSym && sym.typ.equals(t) {
+		if sym, level, ok := sc.lookup(id); ok && level == n.child[i].level && sym.kind == varSym && sym.typ.id() == t.id() {
```

Comment:
as an optimization, maybe we should keep the sym.typ.equals(t) check before doing the sym.typ.id() == t.id() check (instead of completely replacing it)? If the sym.typ.equals(t) is fast(er), and sometimes is enough to return false, then it would allow us to evaluate that the condition is false, without having to do the sym.typ.id() == t.id() one.
But I haven't benchmarked this at all, so I don't know.